### PR TITLE
[IP6NS Grant] Implement support for bind before connect with sync sockets

### PR DESCRIPTION
**Pull request by Kaiepi**
Needed for upcoming support for bind before connect and UDP sockets with `IO::Socket::INET`.

Passes `make test` and `make spectest`.

### CODE REVIEW

File: src/jit/core_templates.expr [hunk 1]
```diff
@@ -20,10 +20,6 @@
 
 (template: set (copy $1))
 
-(template: trunc_i8 (ucast $1 int_sz 1))
```
Comment (jnthn) [inline src/jit/core_templates.expr]: These changes don't seem to belong in this PR?

Comment (Kaiepi) [reply src/jit/core_templates.expr]: These changes are here because I ran the script that sorts the templates so the new listen op would be in the right place.
